code/modules/mob/living/carbon/human/species/vox.dm

```diff
@@ -84,16 +84,21 @@
 /datum/species/vox/after_equip_job(datum/job/J, mob/living/carbon/human/H)
 	if(!H.mind || !H.mind.assigned_role || H.mind.assigned_role != "Clown" && H.mind.assigned_role != "Mime")
 		H.unEquip(H.wear_mask)
-	H.unEquip(H.l_hand)
 
 	H.equip_or_collect(new /obj/item/clothing/mask/breath/vox(H), slot_wear_mask)
 	var/tank_pref = H.client && H.client.prefs ? H.client.prefs.speciesprefs : null
+	var/obj/item/tank/internal_tank
 	if(tank_pref)//Diseasel, here you go
-		H.equip_or_collect(new /obj/item/tank/nitrogen(H), slot_l_hand)
+		internal_tank = new /obj/item/tank/nitrogen(H)
 	else
-		H.equip_or_collect(new /obj/item/tank/emergency_oxygen/vox(H), slot_l_hand)
-	to_chat(H, "<span class='notice'>You are now running on nitrogen internals from the [H.l_hand] in your hand. Your species finds oxygen toxic, so you must breathe nitrogen only.</span>")
-	H.internal = H.l_hand
+		internal_tank = new /obj/item/tank/emergency_oxygen/vox(H)
+	if (!H.equip_to_appropriate_slot(internal_tank))
```

Comment:
No spacing after `if`, [e.g. `if(foo)` vs. `if (bar)`] same with next line please

Reply:
Never thought I could make so many mistakes in such a small PR.
Will fix tomorrow.

Thanks for your all patience!

Reply:
All good man it's just little styling nitpicks

Reply:
Alright, done.